website/src/components/Group/GroupForm.tsx

```diff
@@ -0,0 +1,156 @@
+        const country = formData.get(fieldMapping.country.id) as string;
+        const line1 = formData.get(fieldMapping.line1.id) as string;
+        const line2 = formData.get(fieldMapping.line2.id) as string;
+        const city = formData.get(fieldMapping.city.id) as string;
+        const state = formData.get(fieldMapping.state.id) as string;
+        const postalCode = formData.get(fieldMapping.postalCode.id) as string;
+
+        if (country === chooseCountry) {
+            setErrorMessage('Please choose a country');
+            return false;
+        }
+
+        const group: NewGroup = {
+            groupName,
+            institution,
+            contactEmail,
+            address: { line1, line2, city, postalCode, state, country },
+        };
+
+        const result = await onSubmit(group);
+
+        if (result.succeeded) {
+            window.location.href = result.nextPageHref;
+        } else {
+            setErrorMessage(result.errorMessage);
+        }
+    };
+
+    const isClient = useClientFlag();
+
+    return (
+        <div className='p-4 max-w-6xl mx-auto'>
+            <h2 className='title'>{title}</h2>
+
+            {errorMessage !== undefined && (
+                <ErrorFeedback message={errorMessage} onClose={() => setErrorMessage(undefined)} />
+            )}
+
+            <form onSubmit={internalOnSubmit}>
+                <div className='border-b border-gray-900/10 pb-12 '>
+                    <p className='mt-1 text-sm leading-6 text-gray-600'>
+                        The information you enter on this form will be publicly available on your group page.
+                    </p>
+
+                    <div className='mt-5 grid grid-cols-1 gap-x-6 gap-y-4 sm:grid-cols-6'>
+                        <GroupNameInput defaultValue={defaultGroupData?.groupName} />
+                        <EmailContactInput defaultValue={defaultGroupData?.contactEmail} />
+                        <InstitutionNameInput defaultValue={defaultGroupData?.institution} />
+                        <AddressLineOneInput defaultValue={defaultGroupData?.address.line1} />
+                        <AddressLineTwoInput defaultValue={defaultGroupData?.address.line2} />
+                        <CityInput defaultValue={defaultGroupData?.address.city} />
+                        <StateInput defaultValue={defaultGroupData?.address.state} />
+                        <PostalCodeInput defaultValue={defaultGroupData?.address.postalCode} />
+                        <CountryInput defaultValue={defaultGroupData?.address.country} />
+                    </div>
+
+                    <div className='flex justify-end py-8 gap-4 '>
+                        <button
+                            type='submit'
+                            className='btn btn-primary px-4 py-2 loculusColor text-white rounded'
+                            disabled={!isClient}
+                        >
+                            {buttonText}
```

Comment:
Can we not use some automatic mapping here from field name to id? we can rename html IDs to match field names and use camel case to kebab case. Then this reduces to a few lines instead of 20

Reply:
I'm assuming you mean line 93 to 101 here? Sounds reasonable, I'll look into it!

Now that we're touching the code anyways might be a good time to tidy it up a bit